modulo2/projeto-whatslab/projeto-whatslab/src/components/Whatslabs.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import styledComponents from 'styled-components';
+import { createGlobalStyle } from 'styled-components';
+const GlobalStyled = createGlobalStyle`
+*{
+    margin: 0;
+    padding: 0;
+    background-color: whitesmoke;
+}
+`
+const MainContainer = styledComponents.div`
+display: flex;
+flex-direction: column;
+align-itens: center;
+justify-content: center;
+margin-left: 20%;
+margin-right: 20%;
+background-color: whitesmoke;
+`
+const CaixaDeMensagem = styledComponents.div`
+border: 1px solid gray;
+border-radius: 4px;
+margin-bottom: 10px;
+margin-top: 10px;
+height: 70vh;
+background-color: white;
+
+p{
+    background-color:white;
+}
+`
+const ContainerInputs = styledComponents.div`
+display:flex;
+input {
+    background-color: white;
+    border-style: inset;
+    border-radius: 4px;
+    margin: 5px;
+    width:40%
+}
+input:active{
+    background-color: darkgray;
+}
+button{
+    background-color: white;
+    border-style: inset;
+    border-radius: 4px;
+    margin: 5px;
+    width:20%;
+}
+button:hover{
+    cursor: pointer;
+    background-color:black;
+    color: darkgray;
+}
+`
+class Whatslab extends React.Component {
+
+state = { //state para altereações nos valores dos elemtentos do objeto
+    text: [
+    {
+        nomeUsuario: "",
+        mensagem: ""
+    }
+],
+    inputUser: "",
+    inputText: ""
+};
+ sendText = () => { //função para adicionar usuário, mensagem e zerar o valor do
+     const newMessenger = {
+         nomeUsuario: this.state.inputUser,
+         mensagem: this.state.inputText
+     };
+     const newMessage = [...this.state.text, newMessenger];
+     this.setState({text: newMessage});
+     this.setState({inputText: ""});
+     this.setState({inputUser: ""});
```

Comment:
Aqui você só precisava que um this.setState, nesse caso é melhor usar apenas um porque evitaria renderizações desnecessarias:

`  this.setState({text: newMessage, inputText: "", inputUser: ""}});`